datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -1446,18 +1446,54 @@ impl<'a, S: SimplifyInfo> TreeNodeRewriter for Simplifier<'a, S> {
 
             // Rules for Like
             Expr::Like(Like {
-                expr,
-                pattern,
+                expr: ref like_expr,
+                ref pattern, // Changed to ref pattern to avoid moving
                 negated,
-                escape_char: _,
-                case_insensitive: _,
-            }) if !is_null(&expr)
-                && matches!(
-                    pattern.as_ref(),
-                    Expr::Literal(ScalarValue::Utf8(Some(pattern_str))) if pattern_str == "%"
-                ) =>
-            {
-                Transformed::yes(lit(!negated))
+                escape_char,
+                case_insensitive,
+            }) => {
+                if let Expr::Literal(ScalarValue::Utf8(Some(pattern))) = pattern.as_ref()
+                {
+                    // Special case: pattern is just "%"
+                    if pattern == "%" && !is_null(like_expr) {
+                        return Ok(Transformed::yes(lit(!negated)));
+                    }
+
+                    let pct_wildcard_index = pattern.find('%');
+                    let underscore_index = pattern.find('_');
+
+                    // If no wildcards, treat as equality check
+                    if underscore_index.is_none() && pct_wildcard_index.is_none() {
+                        return Ok(Transformed::yes(Expr::BinaryExpr(BinaryExpr {
+                            left: like_expr.clone(),
+                            op: if negated {
+                                Operator::NotEq
+                            } else {
+                                Operator::Eq
+                            },
+                            right: Box::new(Expr::Literal(ScalarValue::Utf8(Some(
+                                pattern.to_string(),
+                            )))),
+                        })));
+                    }
+
+                    // Handle single % at end case (prefix search)
+                    if let Some(index) = pct_wildcard_index {
+                        if index == pattern.len() - 1 && !case_insensitive && escape_char.is_none() {
+                            let prefix = pattern[..index].to_string();
+                            let new_expr = Expr::ScalarFunction(ScalarFunction {
+                                func: datafusion_functions::string::starts_with(),
```

Comment:
Today a downstream project can choose which functions to use, this is configurable in the session, right? The DF repertoire of functions serves as a convenient package, but not mandatory.
Here we're inserting implicit function call to a specific function.
Is this following a pre-existing pattern? if so, i am OK with this.
But maybe it would be better to eg consult session state whether this function was supposed to be used. What do you think?